packages/scaffold-ui/src/modal/w3m-modal/index.ts

```diff
@@ -210,6 +210,7 @@ export class W3mModal extends LitElement {
   }
 
   private onNewNetwork(nextCaipNetwork: CaipNetwork | undefined) {
+    ApiController.prefetch()
```

Comment:
Do we need to call every API call in the `prefetch` function when changing network, might we want to optimize the requests?
```js
      ApiController.fetchFeaturedWallets(),
      ApiController.fetchRecommendedWallets(),
      ApiController.fetchNetworkImages(),
      ApiController.fetchConnectorImages()
```

Reply:
good point! I'll create a new handler that can do all of this at once 👍

Reply:
There are a couple of edge cases related to this, I¡d like to do it on a separate pr to prevent introdyucing bugs in new release 🙏